src/mixer.c

```diff
@@ -277,7 +278,7 @@ static void SDLCALL
 mix_channels(void *udata, Uint8 *stream, int len)
 {
     Uint8 *mix_input;
-    int i, mixable, volume = MIX_MAX_VOLUME;
+    int i, mixable, volume = master_volume = Mix_MasterVolume(-1);
```

Comment:
It is for reducing of calls to Mix_MasterVolume(-1) and to simplify debbuging. We can call "Mix_MasterVolume" instead of using "master_volume", but in debug it is not convenient to do, and in release mode with optimization a compiler can throw away this variable.

Reply:
Ouch.. you are right, except the fact that `master_volume` in  `mix_channels(...)` hides the global variable `master_volume`. And I mean that compiler can optimize the local variable `master_volume` and can inline call to `Mix_MasterVolume` instead of it.
But anyway, we can remove local variable while the global one is in `mixer.c`.

Reply:
There is no `master_volume` variable in `mix_channels()`, you are just assigning
to the global one.

Anyways, I will push the following shortly:
```diff
diff --git a/src/mixer.c b/src/mixer.c
index 1ebf0cc..5e5710d 100644
--- a/src/mixer.c
+++ b/src/mixer.c
@@ -278,7 +278,7 @@ static void SDLCALL
 mix_channels(void *udata, Uint8 *stream, int len)
 {
     Uint8 *mix_input;
-    int i, mixable, volume = master_volume = Mix_MasterVolume(-1);
+    int i, mixable, volume = Mix_MasterVolume(-1);
     Uint32 sdl_ticks;
 
     (void)udata;
@@ -1643,9 +1643,7 @@ void Mix_UnlockAudio(void)
 
 int Mix_MasterVolume(int volume)
 {
-    int prev_volume;
-
-    prev_volume = master_volume;
+    int prev_volume = master_volume;
     if (volume < 0) {
         return prev_volume;
     }
```

Reply:
Change pushed as https://github.com/libsdl-org/SDL_mixer/commit/ae312b33ad893f803e2a4b6ef6958f74b53f5c3b